test/src/tests-store.ts

```diff
@@ -185,10 +184,158 @@ describe('Store', () => {
 
         expect(store.SDKConfig.sideloadedKits.length, 'side loaded kits').to.equal(sideloadedKits.length);
         expect(store.SDKConfig.sideloadedKits[0], 'side loaded kits').to.deep.equal(sideloadedKit1);
         expect(store.SDKConfig.sideloadedKits[1], 'side loaded kits').to.deep.equal(sideloadedKit2);
     });
+
+    it('should assign apiKey to devToken property', () => {
+        const config = { 
+            ...sampleConfig
+        };
+
+        const store: IStore = new Store(config, window.mParticle.getInstance(), apiKey);
+
+        expect(store.devToken, 'devToken').to.equal(apiKey);
+    });
+
+    describe('#processFlags', () => {
+        it('should return an empty object if no featureFlags are passed', () => {
+            const flags = processFlags({} as SDKInitConfig, {} as SDKConfig);
+            expect(Object.keys(flags).length).to.equal(0);
+        });
+
+        it('should return default featureFlags if no featureFlags are passed', () => {
+            const flags = processFlags({flags: {}} as SDKInitConfig, {} as SDKConfig);
+            const expectedResult = {
+                reportBatching: false,
+                eventBatchingIntervalMillis: 0,
+                offlineStorage: '0',
+                directURLRouting: false,
+            };
+
+            expect(flags).to.deep.equal(expectedResult);
+        });
+
+        it('should return featureFlags if featureFlags are passed in', () => {
+            const cutomizedFlags = {
+                reportBatching: true,
+                eventBatchingIntervalMillis: 5000,
+                offlineStorage: '100',
+                directURLRouting: 'True',
+            };
+
+            const flags = processFlags({flags: cutomizedFlags} as unknown as SDKInitConfig, {} as SDKConfig);
+
+            const expectedResult = {
+                reportBatching: true,
+                eventBatchingIntervalMillis: 5000,
+                offlineStorage: '100',
+                directURLRouting: true,
+            }
+
+            expect(flags).to.deep.equal(expectedResult);
+        });
+    });
+
+    describe('#processBaseUrls', () => {
+        describe('directURLRouting === false', () => {
+            const featureFlags = { directURLRouting: false };
+
+            it('should return default baseUrls if no baseUrls are passed', () => {
+                const baseUrls: Dictionary = Constants.DefaultBaseUrls;
+                const result = processBaseUrls(
+                    {} as unknown as SDKInitConfig,
+                    featureFlags as unknown as IFeatureFlags,
+                    'apikey'
+                );
+
+                expect(result).to.deep.equal(baseUrls)
+            });
+
+            it('should return non-default baseUrls for custom baseUrls that are passed', () => {
+                const config = {
+                    v3SecureServiceUrl: 'testtesttest-custom-v3secureserviceurl/v3/JS/',
+                    configUrl: 'foo-custom-configUrl/v2/JS/',
```

Comment:
Here and below, it would be better to use more realistic URLs, e.g. "foo.customer.com"